### PR TITLE
Feature/dx 1002 correction seamless viewevents

### PR DESCRIPTION
**Pull request by marinafrydenb**

### CODE REVIEW

File: _includes/payment-menu-events.md [hunk 1]
```diff
@@ -0,0 +1,202 @@
+{% assign api_resource = include.api_resource %}
```
Comment (payex-dev) [inline _includes/payment-menu-events.md:1]: Codacy Issue found: [Could not find module `remark-inline-links`](https://app.codacy.com/gh/SwedbankPay/developer.swedbankpay.com/pullRequest?prid=5637621)

**Comment by payex-dev** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/SwedbankPay/developer.swedbankpay.com/pullRequest?prid=5637621&bid=18463042)

File: _includes/payment-menu-events.md [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+described below.
+
+{% else %}
+
+## Seamless View Events
+
+During operation in the Seamless View, several events can occur. They are
+described below.
+
+{% endif %}
```
Comment (asbjornu) [inline _includes/payment-menu-events.md:17]:
```suggestion
    {% assign product="Payment Menu" %}
{% else %}
    {% assign product="Seamless View" %}
{% endif %}

## {{ product }} Events

During operation in the {{ product }}, several events can occur. They are
described below.
```

File: _includes/payment-url.md [hunk 3]
```diff
@@ -8,6 +8,8 @@
     {% assign entity = "Payment" %}
 {% endif %}
 
+## PaymentUrl
```
Comment (asbjornu) [inline _includes/payment-url.md:11]:
```suggestion
## Payment Url
```

File: _includes/payment-url.md [hunk 4]
```diff
@@ -55,15 +57,8 @@ it can lead to issues. If you want to handle payment errors in your own code,
...
-| :--------------------------- | :-------------------------------------------------------------------------------------------------------------------------------------------------- |
-| `onError`                    | Will trigger any time a system error occur from the JavaScript.                                                                                     |
-| `onPaymentFailed`            | will call registrered endpoint if provided. If not subscribed to will redirect to conpleteUrl                                                       |
-| `onPaymentCanceled`          | will be called if the payment is aborted by the consumer or triggered by you with the Abort command on the payment                                  |
-| `onPaymentTransactionFailed` | this will trigger if one payment attempt failed. Standard functionality are to let the consumer be able to retry the payment from the payment page. |
+Events to subscribe to for full control over the payment flow are can be found
+in [Seamless view events][seamless-view-events].
```
Comment (asbjornu) [inline _includes/payment-url.md:61]:
```suggestion
in [Seamless View Events][seamless-view-events].
```

File: checkout/other-features.md [hunk 5]
```diff
@@ -1353,6 +1166,8 @@ The structure of a problem message will look like this:
 
 {% include settlement-reconciliation.md api_resource="paymentorders" %}
 
+{% include payment-menu-events.md api_resource="paymentorders" %}
```
Comment (asbjornu) [inline checkout/other-features.md:1169]: Considering this include is used in both payment instruments and checkout, I think we should go for a more generic name. I think `seamless-view-events` sound more generic than `payment-menu-events`. Can you rename it?

```suggestion
{% include seamless-view-events.md api_resource="paymentorders" %}
```

File: _includes/seamless-view-events.md [hunk 6]
```diff
@@ -0,0 +1,197 @@
+{% assign api_resource = include.api_resource %}
```
Comment (payex-dev) [inline _includes/seamless-view-events.md:1]: Codacy Issue found: [Could not find module `remark-inline-links`](https://app.codacy.com/gh/SwedbankPay/developer.swedbankpay.com/pullRequest?prid=5637621)

**Comment by payex-dev** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/SwedbankPay/developer.swedbankpay.com/pullRequest?prid=5637621&bid=18463042)

**Comment by payex-dev** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/SwedbankPay/developer.swedbankpay.com/pullRequest?prid=5637621&bid=18463042)

File: payments/card/other-features.md [hunk 7]
```diff
@@ -225,10 +116,12 @@ payment"  %}
 [hosted-view]: /payments/card/seamless-view
 [one-click-payments]: #one-click-payments
 [payee-reference]: #payee-reference
+[seamless-view-events]: /payments/card/other-features#seamless-view-events
```
Comment (payex-dev) [inline payments/card/other-features.md:119]: Codacy Issue found: [[no-unused-definitions] Found unused definition](https://app.codacy.com/gh/SwedbankPay/developer.swedbankpay.com/pullRequest?prid=5637621)

File: payments/card/other-features.md [hunk 8]
```diff
@@ -225,10 +116,12 @@ payment"  %}
...
 [split-settlement]: #split-settlement
 [settlement-and-reconciliation]: #settlement-and-reconciliation
 [swedbankpay-support]: https://www.swedbankpay.se/support
 [recurrence]: #recur
 [verify]: #verify
+[verification-flow]: /payments/card/other-features#verification-flow
```
Comment (payex-dev) [inline payments/card/other-features.md:125]: Codacy Issue found: [[no-unused-definitions] Found unused definition](https://app.codacy.com/gh/SwedbankPay/developer.swedbankpay.com/pullRequest?prid=5637621)

**Comment by payex-dev** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/SwedbankPay/developer.swedbankpay.com/pullRequest?prid=5637621&bid=18463042)

File: payments/card/other-features.md (hunk 7 above)
Comment (payex-dev) [inline payments/card/other-features.md:119]: Codacy Issue found: [[no-unused-definitions] Found unused definition](https://app.codacy.com/gh/SwedbankPay/developer.swedbankpay.com/pullRequest?prid=5637621)

File: payments/card/other-features.md (hunk 8 above)
Comment (payex-dev) [inline payments/card/other-features.md:125]: Codacy Issue found: [[no-unused-definitions] Found unused definition](https://app.codacy.com/gh/SwedbankPay/developer.swedbankpay.com/pullRequest?prid=5637621)

File: payments/card/other-features.md (hunk 7 above)
Comment (asbjornu) [reply payments/card/other-features.md:119]: @marinafrydenb ☝️

File: payments/card/other-features.md (hunk 8 above)
Comment (asbjornu) [reply payments/card/other-features.md:125]: @marinafrydenb ☝️

**Comment by payex-dev** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 5
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/SwedbankPay/developer.swedbankpay.com/pullRequest?prid=5637621&bid=18463042)

File: payments/swish/other-features.md [hunk 9]
```diff
@@ -540,8 +538,11 @@ Content-Type: application/json
...
 ```
 
+{% include seamless-view-events.md api_resource="swish" %}
+
 {% include iterator.html prev_href="after-payment" prev_title="Back: After
 Payment" %}
 
 [payee-reference]: #payee-reference
 [transaction-resource]: #Transactions
+[seamless-view-events]: #seamless-view-events
```
Comment (payex-dev) [inline payments/swish/other-features.md:548]: Codacy Issue found: [[no-unused-definitions] Found unused definition](https://app.codacy.com/gh/SwedbankPay/developer.swedbankpay.com/pullRequest?prid=5637621)

File: payments/vipps/other-features.md [hunk 10]
```diff
@@ -272,3 +272,4 @@ All Vipps error types will have the following URI in front of type:
 [technical-reference-expansion]: /home/technical-information#expansion
 [user-agent]: https://en.wikipedia.org/wiki/User_agent
 [technical-reference-payeeReference]: #payee-reference
+[seamless-view-events]: #seamless-view-events
```
Comment (payex-dev) [inline payments/vipps/other-features.md:275]: Codacy Issue found: [[no-unused-definitions] Found unused definition](https://app.codacy.com/gh/SwedbankPay/developer.swedbankpay.com/pullRequest?prid=5637621)

File: payments/card/other-features.md [hunk 11]
```diff
@@ -230,5 +121,6 @@ payment"  %}
 [swedbankpay-support]: https://www.swedbankpay.se/support
 [recurrence]: #recur
 [verify]: #verify
+[seamless-view-events]: #seamless-view-events
```
Comment (payex-dev) [inline payments/card/other-features.md:124]: Codacy Issue found: [[no-unused-definitions] Found unused definition](https://app.codacy.com/gh/SwedbankPay/developer.swedbankpay.com/pullRequest?prid=5637621)

File: payments/invoice/other-features.md [hunk 12]
```diff
@@ -583,5 +585,6 @@ Payment" %}
 [recur]: #recur
 [redirect]: /payments/invoice/redirect
 [seamless-view]: /payments/invoice/seamless-view
+[seamless-view-events]: #seamless-view-events
```
Comment (payex-dev) [inline payments/invoice/other-features.md:588]: Codacy Issue found: [[no-unused-definitions] Found unused definition](https://app.codacy.com/gh/SwedbankPay/developer.swedbankpay.com/pullRequest?prid=5637621)

File: checkout/other-features.md [hunk 13]
```diff
@@ -1384,6 +1200,7 @@ elements. When in doubt, please follow the [robustness principle](https://en.wik
 [order-items]: /checkout/other-features#order-items
 [payee-reference]: /checkout/other-features#payee-reference
 [payment-menu]: /checkout/payment-menu
+[payment-menu-events]: #payment-menu-events
```
Comment (payex-dev) [inline checkout/other-features.md:1203]: Codacy Issue found: [[no-unused-definitions] Found unused definition](https://app.codacy.com/gh/SwedbankPay/developer.swedbankpay.com/pullRequest?prid=5637621)

File: _includes/seamless-view-events.md [hunk 14]
```diff
@@ -0,0 +1,197 @@
...
+{:.table .table-striped}
+| Field        | Type     | Description                                                                                                                                                      |
+| :----------- | :------- | :--------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `name`       | `string` | The name and identifier of specific instrument instances - i.e. if you deploy more than one type of credit card payments, they would be distinguished by `name`. |
+| `instrument` | `string` | `Creditcard`, `vipps`, `swish`, `invoice`. The instrument selected by the user.                                                                                  |
+
+### `onPaymentCreated`
+
+This event triggers when a user has selected a payment instrument and actively
+attempts to perform a payment. The `onPaymentCreate` event is raised with the
```
Comment (asbjornu) [inline _includes/seamless-view-events.md:154]:
```suggestion
attempts to perform a payment. The `onPaymentCreated` event is raised with the
```

File: _includes/seamless-view-events.md [hunk 15]
```diff
@@ -0,0 +1,197 @@
...
+following event argument object:
+
+{:.code-header}
+**`onPaymentCreated` event object**
+
+```js
+{
+    "id": "/psp/{{ api_resource }}/payments/{{ page.payment_id }}",
+    "instrument": "creditcard | vipps | swish | invoice",
```
Comment (asbjornu) [inline _includes/seamless-view-events.md:163]: No pipes in code examples, please.

```suggestion
    "instrument": "creditcard",
```

**Pull request merged**